src/CqlBuilderPanel/functionsSection/functions/Functions.tsx

```diff
@@ -62,11 +68,20 @@ const Functions = ({
     setCurrentLimit(e.target.value);
     setCurrentPage(1);
   };
-
   useEffect(() => {
     managePagination();
   }, [functions, currentPage, currentLimit]);
 
+  // edit dialog utilities
+  const [discardDialog, setDiscardDialog] = useState({
+    open: false,
+    operation: null,
+  });
+  const [editFunctionDialogOpen, setEditFunctionDialogOpen] =
+    useState<boolean>();
+  const [deleteDialogOpen, setDeleteDialogOpen] = useState<boolean>(false);
+  const [selectedFunction, setSelectedFunction] = useState<Lookup>();
```

Comment:
FunctionLookup actually extends Lookup, do you think it's ok to use FunctionLookup instead? On Line 164 with setSelectedFunction(), table.getRow(row.cell.row.id).original has all attributes of FunctionLookup.